src/wp-includes/blocks/footnotes.php

```diff
@@ -78,3 +82,137 @@ function register_block_core_footnotes() {
 	);
 }
 add_action( 'init', 'register_block_core_footnotes' );
+
+add_action(
+	'wp_after_insert_post',
+	/**
+	 * Saves the footnotes meta value to the revision.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	static function( $revision_id ) {
+		$post_id = wp_is_post_revision( $revision_id );
+
+		if ( $post_id ) {
+			$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+			if ( $footnotes ) {
+				// Can't use update_post_meta() because it doesn't allow revisions.
+				update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+			}
+		}
+	}
+);
+
+add_action(
+	'_wp_put_post_revision',
+	/**
+	 * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	static function( $revision_id ) {
+		global $_gutenberg_revision_id;
```

Comment:
What about the `gutenberg` prefix here. Is that okay?

Reply:
Also quite a hacky workaround in general having to use such a global here :/

Reply:
I understand this is a temporary situation until revisions properly support post meta in core.

Reply:
Open for any better workarounds. It's quite late now to be adding general post meta revision support or start changing the REST API hook order.